OPAL/br/src/main/scala/org/opalj/br/analyses/cg/EntryPointFinder.scala

```diff
@@ -347,4 +347,46 @@ object AllEntryPointsFinder extends EntryPointFinder {
             project.allProjectClassFiles.flatMap(_.methodsWithBody)
         else project.allMethodsWithBody
     }
+}
+
+/**
+ * The AndroidEntryPointFinder considers specific methods of app components as entry points.
+ * It does not work for androidx
+ *
+ * @author Tom Nikisch
+ */
+object AndroidEntryPointsFinder extends EntryPointFinder {
+
+    val activityEPS: List[String] = List("onCreate", "onRestart", "onStart", "onResume",
+        "onStop", "onDestroy", "onActivityResult")
+    val serviceEPS: List[String] = List("onCreate", "onStartCommand", "onBind", "onStart")
+    val contentProviderEPS: List[String] = List("onCreate", "query", "insert", "update")
+    val locationListenerEPS: List[String] = List("onLocationChanged", "onProviderDisabled", "onProviderEnabled",
+        "onStatusChanged")
+    val onNmeaMessageListenerEPS: List[String] = List("onNmeaMessage")
+    val broadcastReceiverEPS: List[String] = List("onReceive")
+    val defaultEPS = Map("android/app/Activity" -> activityEPS, "android/app/Service" -> serviceEPS,
+        "android/content/ContentProvider" -> contentProviderEPS,
+        "android/location/LocationListener" -> locationListenerEPS,
+        "android/location/onNmeaMessageListener" -> onNmeaMessageListenerEPS,
+        "android/content/BroadcastReceiver" -> broadcastReceiverEPS)
+
+    override def collectEntryPoints(project: SomeProject): Traversable[Method] = {
+        val eps = ArrayBuffer.empty[Method]
+        for ((superClass, methodList) ← defaultEPS) {
+            eps ++= findEPS(ObjectType(superClass), methodList, project)
+        }
+        eps
+    }
+
+    def findEPS(ot: ObjectType, possibleEPS: List[String], project: SomeProject): ArrayBuffer[Method] = {
+        val eps = ArrayBuffer.empty[Method]
+        val classHierarchy = project.classHierarchy
+        classHierarchy.foreachSubclass(ot, project) { sc ⇒
+            for (pep ← possibleEPS; m ← sc.findMethod(pep) if m.body.isDefined && !eps.contains(m)) {
```

Comment:
Just to be sure: Are you aware that findMethod with only a method name will return all methods of the given name? That is probably fine if the methods that you are looking for can come with different signatures for the same name and still be entry points to the call graph and if it is unlikely that someone will implement a method with one of these names that is not an entry point. On the other hand, if there is only one (or in few cases maybe two or three) possible signature anyway and if it is conceivable that someone would have non-entry point methods of those names, I would suggest to also include these signatures in the query here.